main/src/ca/uwaterloo/flix/language/ast/TokenKind.scala

```diff
@@ -17,18 +17,323 @@ package ca.uwaterloo.flix.language.ast
 
 sealed trait TokenKind
 
+// NOTE: Tokens are named for 'what they are' rather than 'what they represent'.
+// So '::' is not named 'Cons' but instead 'ColonColon' as the lexer should be oblivious to the concept of cons
+
+// NOTE: Builtin type keywords like 'Float32' are lexed into 'Float32Keyword' whereas Float32 *literals* are lexed to 'Float32'
```

Comment:
KeywordFloat32

Reply:
I don't think we need to lex `Int32`. and I think we want to call the literal `LiteralInt32`.